finetune/base_models/gpt2/encoder.py

```diff
@@ -117,35 +124,47 @@ def bpe(self, token):
                 break
             else:
                 pairs = get_pairs(word)
-        word = ' '.join(word)
+        word = " ".join(word)
         self.cache[token] = word
         return word
 
-    def _encode(self, texts, labels=None):
+    def _encode(self, texts, labels=None, context=None):
         """
-        Convert a batch of raw text to a batch of byte-pair encoded token indices.
+        Convert a sample of raw text to a list of byte-pair encoded token indices.
         """
         self._lazy_init()
-
         batch_tokens = []
         batch_token_idxs = []
         batch_label_idxs = []
         batch_character_locs = []
+        batch_original_character_locs = (
+            []
+        )  # to account for the fact that some BPEs have different lengths than their original tokens (e.g. special characters such as bullets)
+        batch_context = []
         label = None
+        context_ = None
+        offset = (
```

Comment:
Doesn't need to be a tuple.  Same with batch_original_character_locs.

Reply:
Well I guess this doesn't even make a tuple.  But the parens aren't used

Reply:
^ Yeah that's some black formatting fun